src/Microsoft.AspNet.Mvc.Razor/Compilation/ExpressionRewriter.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                return node;
+            }
+
+            if (!node.Parent.IsKind(SyntaxKind.Argument))
+            {
+                return node;
+            }
+
+            var parameter = node.Parameter;
+            if (IsValidForHoisting(parameter, node.Body))
+            {
+                // Replace with a MemberAccess
+                var memberName = string.Format(FieldNameTemplate, Expressions.Count);
+                var memberAccess = PadMemberAccess(node, SyntaxFactory.IdentifierName(memberName));
+                Expressions.Add(new KeyValuePair<SimpleLambdaExpressionSyntax, IdentifierNameSyntax>(node, memberAccess));
+                return memberAccess;
+            }
+
+            return node;
+        }
+
+        private IdentifierNameSyntax PadMemberAccess(SimpleLambdaExpressionSyntax node, IdentifierNameSyntax memberAccess)
+        {
+            // We want to make the new span 
+            var originalSpan = node.GetLocation().GetMappedLineSpan();
+
+            // Start by collecting all the trivia 'inside' the expression - we need to tack that on the end, but
+            // if it ends with a newline, don't include that.
+            var innerTrivia = SyntaxFactory.TriviaList(node.DescendantTrivia(descendIntoChildren: n => true));
+            if (innerTrivia.Count > 0 && innerTrivia[innerTrivia.Count - 1].IsKind(SyntaxKind.EndOfLineTrivia))
+            {
+                innerTrivia = innerTrivia.RemoveAt(innerTrivia.Count - 1);
+            }
+
+            memberAccess = memberAccess.WithTrailingTrivia(innerTrivia);
+
+            // If everything is all on one line, then make sure the spans are the same, to compensate
+            // for the expression potentially being longer than the variable name.
+            var lineSpan = originalSpan.EndLinePosition.Line - originalSpan.StartLinePosition.Line;
+            if (lineSpan == 0)
+            {
+                var padding = node.Span.Length - memberAccess.FullSpan.Length;
+                var trailingTrivia = 
+                    SyntaxFactory.TriviaList(memberAccess.GetTrailingTrivia())
+                    .Add(SyntaxFactory.Whitespace(new string(' ', padding)))
+                    .AddRange(node.GetTrailingTrivia());
+
+                return 
+                    memberAccess
+                    .WithLeadingTrivia(node.GetLeadingTrivia())
+                    .WithTrailingTrivia(trailingTrivia);
+            }
+            else
+            {
+                // If everything isn't on the same line, we need to pad out the last line.
+                var padding =
+                    originalSpan.EndLinePosition.Character -
+                    originalSpan.StartLinePosition.Character;
+
+                var trailingTrivia =
+                    SyntaxFactory.TriviaList(memberAccess.GetTrailingTrivia())
+                    .Add(SyntaxFactory.Whitespace(new string(' ', padding)))
```

Comment:
Tabs intermingled with the code would be handled by L98. The padding that we manufacture is only there to accommodate the difference in spans between our expression and 'non-trivia' code.